contents/node/nodejs.md

```diff
@@ -0,0 +1,224 @@
+}
+```
+Let's look at some of its components. Every package/app needs a **name** and a **version**. Together, the name and version should form a unique identifier. When the package is updated, the version number must be updated as well.
+
+Adding a good **description** string and **keywords** e.g. `["promise", "lock"]` will help others to discover your package.
+
+There are other common properties in `package.json`
+
+### respository
+
+```json
+{
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/ashleygwilliams/my_package.git"
+  }
+}
+```
+The `repository` property defines where the source code for the module lives. It will be useful for people to locate your source code and even contribute to it.
+
+### dependencies and devDependencies
+
+The `dependencies` property specifies dependencies needed in production while the `devDependencies` property specifies dependencies needed in development.
+
+```json
+{
+  "dependencies": {
+    "express": "~3.0.1",
+    "sequelize":"latest",
+    "bluebird": "^3.4.7",
+    "angular":"latest",
+    "angular-ui-router": "~0.2.11",
+    "path":"latest",
+  },
+  "devDependencies": {
+    "eslint": "^4.16.0",
+    "eslint-config-airbnb-base": "^12.1.0",
+    "eslint-plugin-import": "^2.8.0"
+  }
+}
+```
+The `dependencies` property maps to an object that has the name and version range for each dependency. It is common to find carets (`^`) and tildes (`~`) in the version range. To understand what they are for, checkout [NodeSource's article on semvers](https://nodesource.com/blog/semver-a-primer/).
+
+It is also possible to install a new dependency and update `package.json` directly from the command line with
+```
+$ npm install <package_name> --save
+```
+For dependencies only needed in development, use
+```
+$ npm install <package_name> --save-dev
+```
+When a dependency is installed, the package's code will be added to the local `/node_modules` folder.
+
+You can find a wide range of useful packages on [npmjs](https://www.npmjs.com/). See the [module section](#node-module-system) for how to import packages/modules in your code.
+
+### scripts
+
+The `scripts` property is a useful tool that can be used to perform repetitive tasks such as testing and building. It takes in an object. For each property, the key is the given name of the command and the corresponding value is the actual command that will be run.
+
+For example, inside your package.json, you might have
+```json
+{
+  "scripts": {
+      "build": "node app.js",
+      "lint": "eslint **/*.js",
+      "lint-fix": "eslint --fix **/*.js"
+  }
+}
+```
+Running `npm run build` in the command shell will execute `node app.js` and similarly `npm run lint-fix` will fix your linting errors in your JavaScript files if you have eslint installed. To find out more about how you can use npm scripts, refer to [freeCodeCamp's Introduction to NPM Scripts](https://medium.freecodecamp.org/introduction-to-npm-scripts-1dbb2ae01633).
```

Comment:
Should add `$` to all the terminal commands in this line.

Reply:
I thought it was weird to add it inline since it says runs x in the command shell